### PR TITLE
Don't throw exceptions for missing windowing platform since it's actually not mandatory for TrayIcon

### LINKED ISSUE #8603: TrayIcon on ISingleViewApplication
**Describe the bug**
When running ControlCatalog on `ISingleViewApplicationLifetime` app crashes on start because `IWindowingPlatform` is **not registered** on `PlatformManager.CreateTrayIcon()`.

`
Unhandled exception. System.Exception: Could not CreateTrayIcon(): IWindowingPlatform is not registered.
   at Avalonia.Controls.Platform.PlatformManager.CreateTrayIcon() in /Users/wenzl/RiderProjects/Avalonia/src/Avalonia.Controls/Platform/PlatformManager.cs:line 29
`

**To Reproduce**

1. Run ControlCatalog.NetCore on Framebuffer or Drm.

**Expected behavior**
Working app.

**Screenshots**

**Desktop (please complete the following information):**

- OS: TDX Wayland with XWayland 5.7.0-devel-20220627150124+build.0 (dunfell)
- Version 0.10.17

### LINKED ISSUE #8605: WindowIcon on ISingleViewAppliction
**Describe the bug**
When running ControlCatalog on ISingleViewApplicationLifetime app crashes on start because **IPlatformIconLoader** is not locate able on WindowIcon`.

`Unhandled exception. System.InvalidOperationException: Unable to locate 'Avalonia.Platform.IPlatformIconLoader'.
   at Avalonia.LocatorExtensions.GetRequiredService[T](IAvaloniaDependencyResolver resolver) in /Users/wenzl/RiderProjects/Avalonia/src/Avalonia.Base/AvaloniaLocator.cs:line 136
   at Avalonia.Controls.WindowIcon..ctor(Stream stream) in /Users/wenzl/RiderProjects/Avalonia/src/Avalonia.Controls/WindowIcon.cs:line 24`

**To Reproduce**

1. Run ControlCatalog.NetCore on Framebuffer or Drm.

**Expected behavior**
Working app.

**Desktop (please complete the following information):**

OS: TDX Wayland with XWayland 5.7.0-devel-20220627150124+build.0 (dunfell)
Version 0.10.17

### PR DESCRIPTION
**Pull request by kekekeks**
Should fix #8603

### CODE REVIEW

File: src/Avalonia.Controls/Platform/PlatformManager.cs [hunk 1]
```diff
@@ -25,9 +25,7 @@ public static void SetDesignerScalingFactor(double factor)
             var platform = AvaloniaLocator.Current.GetService<IWindowingPlatform>();
 
             if (platform == null)
-            {
-                throw new Exception("Could not CreateTrayIcon(): IWindowingPlatform is not registered.");
-            }
+                return null;
 
             return s_designerMode ? null : platform.CreateTrayIcon();
```
Comment (jp2masa) [inline src/Avalonia.Controls/Platform/PlatformManager.cs:30]: Maybe simplify this to `return s_designerMode ? null : platform?.CreateTrayIcon();` and remove the `if`?

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022514-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022518-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022523-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by timunie** [comment]
@kekekeks FYI: "Should fix https://github.com/AvaloniaUI/Avalonia/issues/8605" doesn't seem to actually close the issue, if that was intended.

**Pull request merged**